tools/rest_multi_models.lua

```diff
@@ -0,0 +1,251 @@
+   },
+   {
+     '-model_config', '',
+     [[Path to yaml configuration file.]],
+     {
+       valid = onmt.utils.ExtendedCmdLine.fileExists
+     }
+   }
+}
+
+cmd:setCmdLineOptions(server_options, 'Server')
+onmt.utils.Cuda.declareOpts(cmd)
+onmt.utils.Logger.declareOpts(cmd)
+
+
+local opt_server = cmd:parse(arg)
+local fconfig = io.open(opt_server.model_config, "rb")
+assert(fconfig)
+local content_config = fconfig:read "*a"
+fconfig:close()
+
+local server_cfg = yaml.load(content_config)
+
+-- CAREFULL opt is a table of options here, ie a table of what opt is usually in onmt
+
+local opt = {}
+
+for i=1, #server_cfg do
+  local model_cmd = onmt.utils.ExtendedCmdLine.new()
+  local model_config = {}
+  -- if withAttn is not defined at model level
+  if not server_cfg[i]['-withAttn'] then
+    table.insert(model_config, {'-withAttn', opt_server.withAttn, ''})
+  end
+  for k,v in pairs(server_cfg[i]) do
+    table.insert(model_config, {'-'..k, v, ''})
+  end
+  model_cmd:setCmdLineOptions(model_config, 'Server')
+  onmt.translate.Translator.declareOpts(model_cmd)
+  tokenizer.declareOpts(model_cmd)
+
+  model_cmd:text("")
+  model_cmd:text("**Other options**")
+  model_cmd:text("")
+  model_cmd:option('-batchsize', 1000, [[Size of each parallel batch - you should not change except if low memory.]])
+  opt[i] = model_cmd:parse({})
+end
+
+
+local function translateMessage(server,lines)
+  local batch = {}
+ -- We need to tokenize the input line before translation
+  local bpe
+  local res
+  local err
+ -- first item contains both the src (to translate) AND the id of the engine
+ -- these local variables are set to match the previous version of code
+  local options = server.opt[lines[1].id]
+  local translator = server.translator[lines[1].id]
+
+  _G.logger:info("Start Tokenization")
+  if options.bpe_model ~= '' then
+     bpe = BPE.new(options)
+  end
+  for i = 1, #lines do
+    local srcTokenized = {}
+    local tokens
+    local srcTokens = {}
+    res, err = pcall(function() tokens = tokenizer.tokenize(options, lines[i].src, bpe) end)
+     -- it can generate an exception if there are utf-8 issues in the text
+     if not res then
+       if string.find(err, "interrupted") then
+         error("interrupted")
+        else
+         error("unicode error in line " .. err)
+       end
+     end
+    table.insert(srcTokenized, table.concat(tokens, ' '))
+    -- Extract from the line.
+    for word in srcTokenized[1]:gmatch'([^%s]+)' do
+      table.insert(srcTokens, word)
+    end
+    -- Currently just a single batch.
+    table.insert(batch, translator:buildInput(srcTokens))
+  end
+  -- Translate
+  _G.logger:info("Start Translation")
+  local results = translator:translate(batch)
+  _G.logger:info("End Translation")
+
+  -- Return the nbest translations for each in the batch.
+  local translations = {}
+  for b = 1, #lines do
+    local ret = {}
+    for i = 1, translator.opt.n_best do
```

Comment:
we need to change this .opt. by .args. It was changed in a previous commit in master. I had it on my local changes but never pushed it.